src/steamship/agents/functions_based.py

```diff
@@ -0,0 +1,127 @@
+from typing import Iterable, List, Optional
+
+from steamship import Block, MimeTypes, SteamshipError, Tag
+from steamship.agents.llms.steamship_llm import SteamshipLLM
+from steamship.agents.schema import Action, Agent, AgentContext, FinishAction, Tool
+from steamship.agents.utils import build_chat_history
+from steamship.data.tags.tag_constants import ChatTag, RoleTag, TagKind, TagValueKey
+from steamship.data.tags.tag_utils import get_tag
+from steamship.plugin.capabilities import (
+    ConversationSupport,
+    FunctionCallingSupport,
+    SystemPromptSupport,
+)
+
+
+class FunctionsBasedAgent(Agent):
+    """Selects actions for AgentService based on a set of Tools."""
+
+    PROMPT = """You are a helpful AI assistant.
+
+NOTE: Some functions return images, video, and audio files. These multimedia files will be represented in messages as
+UUIDs for Steamship Blocks. When responding directly to a user, you SHOULD print the Steamship Blocks for the images,
+video, or audio as follows: `Block(UUID for the block)`.
+
+Example response for a request that generated an image:
+Here is the image you requested: Block(288A2CA1-4753-4298-9716-53C1E42B726B).
+
+Only use the functions you have been provided with."""
+
+    def __init__(self, llm: SteamshipLLM, tools: List[Tool], **kwargs):
```

Comment:
should we use something like `warnings.warn("not ready", RuntimeWarning, ...)` or whatever is appropriate here, until we have tests and examples converted?

Reply:
What do you think about just `_` prefixing these?